Dockerfile.rocm.ubi

```diff
@@ -139,132 +113,103 @@ RUN --mount=type=cache,target=${CCACHE_DIR} \
+    --mount=type=cache,target=/root/.cache/pip \
+    --mount=type=cache,target=/root/.cache/uv \
     if [ "$BUILD_TRITON" = "1" ]; then \
         mkdir -p /libs && cd /libs && \
         git clone https://github.com/OpenAI/triton.git && \
         cd triton && \
         git checkout "${TRITON_BRANCH}" && \
         cd python && \
+        uv pip install -v cmake ninja && \
         python3 setup.py bdist_wheel --dist-dir=/install; \
     else \
-        # Create an empty directory otherwise as later build stages expect one
-        mkdir -p /install; \
+        mkdir -p /install && \
+        wget -P /install "https://download.pytorch.org/whl/nightly/pytorch_triton_rocm-3.0.0%2B${TRITON_WHEEL_SHA}-cp311-cp311-linux_x86_64.whl"; \
     fi
 
 ##################################################################################################
 
-### Final vLLM build stage
-FROM build_base AS final
-
-# Import the vLLM development directory from the build context
-COPY . .
+FROM rocm_devel AS build_vllm
+ARG PYTORCH_ROCM_ARCH="gfx908;gfx90a;gfx942;gfx1100"
+ARG MAX_JOBS
+ENV MAX_JOBS=${MAX_JOBS}
 
-# Install wget only if it is needed
-RUN microdnf -y install wget && microdnf clean all
 
-# Package upgrades to avoid dependency issues and add functionality
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python3 -m pip install --upgrade numba scipy huggingface-hub[cli] && \
-    microdnf clean all
+COPY . .
 
-ARG PYTORCH_ROCM_ARCH="gfx908;gfx90a;gfx942;gfx1100"
 
-# Set environment variables for runtime
-ENV RAY_EXPERIMENTAL_NOSET_ROCR_VISIBLE_DEVICES=1
-# Silences the HF Tokenizers warning
-ENV TOKENIZERS_PARALLELISM=false
+ENV VLLM_TARGET_DEVICE="rocm"
+ENV MAX_JOBS=${MAX_JOBS}
+# Make sure punica kernels are built (for LoRA)
+ENV VLLM_INSTALL_PUNICA_KERNELS=1
 
-# Install dependencies from requirements file and apply ROCm specific patches
-RUN --mount=type=cache,target=${CCACHE_DIR} \
+RUN --mount=type=cache,target=/root/.cache/ccache \
     --mount=type=cache,target=/root/.cache/pip \
-    python3 -m pip install -Ur requirements-rocm.txt && \
-    ROCM_VERSION=$(ls /opt | grep -Po 'rocm-[0-9]+\.[0-9]+') && \
-    case "$ROCM_VERSION" in \
-        "rocm-6.1") \
-            # Apply patch for ROCm 6.1
-            wget -N https://github.com/ROCm/vllm/raw/fa78403/rocm_patch/libamdhip64.so.6 -P /opt/rocm/lib && \
-            # Remove potentially conflicting HIP runtime from torch
-            rm -f "$(python3 -c 'import torch; print(torch.__path__[0])')"/lib/libamdhip64.so* || true ;; \
-        *) \
-            echo "ROCm version $ROCM_VERSION is not supported for patching." ;; \
-    esac && \
+    --mount=type=cache,target=/root/.cache/uv \
+    uv pip install -v -U \
+        --extra-index-url "https://download.pytorch.org/whl/nightly/rocm6.1" \
+        --index-strategy=unsafe-best-match \
+        ninja "cmake>=3.26" packaging \
+        torch==2.5.0.dev20240726+rocm6.1 \
+        torchvision==0.20.0.dev20240726+rocm6.1 && \
```

Comment:
we already installed torch at line 77, do we just copy files from mounted cache?

Reply:
Moved the torch install to the `rocm_base` layer instead